### PR TITLE
AutoSuggestBox - Use default control templates for WinUI3

### LINKED ISSUE #5144: Auto Suggest Box - Use default control templates for WinUI3

### PR DESCRIPTION
**Pull request by buyaa-n**
### AutoSuggestBox - Use default control templates for WinUI3

1. Mapped supported properties with WinUI AutoSuggestBox
2. Removed `MauiAutoSuggestBoxStyle.xaml and `MauiSearchTextBox.cs`
3. Added background color mapping for windows
4. Removed Clear button support from windows

Before:
SearchBar_before

After:
image

### Issues Fixed

Fixes # https://github.com/dotnet/maui/issues/5144

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 1]
```diff
@@ -16,110 +35,85 @@ public static void UpdatePlaceholder(this AutoSuggestBox platformControl, ISearc
...
+				{
+					resource.Remove(key);
+				}
+			}
+			else
+			{
+				foreach (var key in keys)
+				{
+					resource[key] = brush;
+				}
```
Comment (antonfirsov) [inline src/Core/src/Platform/Windows/SearchBarExtensions.cs:84]: Nit: #5479 added utility methods `ResourceDictionaryExtensions.RemoveKeys/SetValueForAllKey` for these.

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 2]
```diff
@@ -16,110 +35,85 @@ public static void UpdatePlaceholder(this AutoSuggestBox platformControl, ISearc
...
+		public static void UpdateIsReadOnly(this AutoSuggestBox platformControl, ISearchBar searchBar)
 		{
-			if (queryTextBox == null)
-				queryTextBox = platformControl.GetFirstDescendant<MauiSearchTextBox>();
-
-			if (queryTextBox == null)
-				return;
-
-			queryTextBox.IsTextPredictionEnabled = searchBar.IsTextPredictionEnabled;
+			platformControl.IsEnabled = searchBar.IsReadOnly;
```
Comment (buyaa-n) [inline src/Core/src/Platform/Windows/SearchBarExtensions.cs:111]: `ReadOnly` also not supported/exist on `AutoSuggestBox`, assumed `IsEnabled` could match, but SearchBar has `IsEnabled` property which is also mapped, maybe remove this mapping CC @mattleibow

**Comment by Redth** [comment]
This fixes #5144

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs (hunk 2 above)
Comment (Redth) [reply src/Core/src/Platform/Windows/SearchBarExtensions.cs:105]: I think this is a good approach for now.

**Review by Redth — APPROVED** [review]

**Pull request merged**